src/background/services/openPayments.ts

```diff
@@ -424,31 +424,46 @@ export class OpenPaymentsService {
     if (calculatedHash !== hash) throw new Error('Invalid interaction hash')
   }
 
-  private async confirmPayment(url: string): Promise<InteractionParams> {
+  private async closeTab(currentTab: number, tabToClose: number) {
+    await this.browser.tabs.update(currentTab, { active: true })
+    await this.browser.tabs.remove(tabToClose)
+  }
+
+  private async getInteractionInfo(url: string): Promise<InteractionParams> {
     const currentTab = await getCurrentActiveTab(this.browser)
 
     return await new Promise((res) => {
-      if (url) {
-        this.browser.tabs.create({ url }).then((tab) => {
-          if (tab.id) {
-            this.browser.tabs.onUpdated.addListener((tabId, changeInfo) => {
-              try {
-                const tabUrl = new URL(changeInfo.url || '')
-                const interactRef = tabUrl.searchParams.get('interact_ref')
-                const hash = tabUrl.searchParams.get('hash')
-
-                if (tabId === tab.id && interactRef && hash) {
-                  this.browser.tabs.update(currentTab.id, { active: true })
-                  this.browser.tabs.remove(tab.id)
-                  res({ interactRef, hash })
-                }
-              } catch (e) {
-                /* do nothing */
+      this.browser.tabs.create({ url }).then((tab) => {
+        if (tab.id) {
+          const getInteractionInfo: Parameters<
+            Tabs.onUpdatedEvent['addListener']
+          >[0] = async (tabId, changeInfo) => {
+            if (tabId !== tab.id) return
+            try {
```

Comment:
Just one nit to avoid the extra-indented code.
```ts
if (!tab.id) return
// instead of if (tab.id) { ... }
```

and maybe also
```ts
type TabUpdateCallback = Parameters<Tabs.onUpdatedEvent['addListener']>[0]
```